### PR TITLE
Overflow chip with reduced column usage

### PR DESCRIPTION
**Pull request by sifnoc**
Modified overflow chip to utilize only two columns, one for `value` and the other one for `decomposed_values`.

### CODE REVIEW

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (enricobottazzi) [inline zk_prover/src/chips/overflow/overflow_check.rs:37]: It may end up that num_rows is not an integer number if MAX_BITS is an odd number. I don't know if there's a situation in which makes sense to use odd `MAX_BITS`, maybe we can add an assert to avoid this.

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (enricobottazzi) [inline zk_prover/src/chips/overflow/overflow_check.rs:9]: I suggest adding this as generic to the Config too, so one can modify the range check freely

**Review by enricobottazzi — CHANGES_REQUESTED** [review]
The PR LGTM after these small changes. After #67 gets merged, I'd like to explore other ways of designing this that may lead to more efficiency. For example spreading the values across the 4 advice columns that we have at our disposal, in order to reduce the number of rows used and the relative offset!

**Review by alxkzmn — APPROVED** [review]

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (sifnoc) [reply zk_prover/src/chips/overflow/overflow_check.rs:37]: The num_rows is currently calculated as an integer, regardless of whether `MAX_BITS` is an odd number. |
However, I think it's better to add a warning sign for when `MAX_BITS` is an odd number, as future developers might expect different results.

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (sifnoc) [reply zk_prover/src/chips/overflow/overflow_check.rs:9]: Ah, it makes sense. I will fix and revert 16bits test cases.

Comment (sifnoc) [reply zk_prover/src/chips/overflow/overflow_check.rs]: Fixed at [7da5277](https://github.com/summa-dev/summa-solvency/pull/68/commits/7da5277dac72057e94354e80442d60ee1b63bb7a)

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (sifnoc) [reply zk_prover/src/chips/overflow/overflow_check.rs:48]: Added Warn at [ffce080](https://github.com/summa-dev/summa-solvency/pull/68/commits/ffce0801a8248d428b00d7e0c954c8a2ee794fc4)

**Review by enricobottazzi — APPROVED** [review]
🚀

File: zk_prover/src/chips/overflow/overflow_check.rs
Comment (sifnoc) [reply zk_prover/src/chips/overflow/overflow_check.rs:48]: this commit checks whether the `MOD_BITS` was divided evenly by `MAX_BITS`

**Pull request merged**
